### PR TITLE
Tests and a bug fix

### PR DESCRIPTION
**Pull request by talvarez0507**
This fixes an issue we have in master where tasks are not being sent correctly to the servlet and the button has the wrong function. It also adds tests to for some java methods.

### CODE REVIEW

**Review by hollyyuqizheng — APPROVED** [review]

File: portfolio/src/main/webapp/task.js [hunk 1]
```diff
@@ -231,8 +231,7 @@ function collectAllTasks() {
         taskDescription,
         getDurationMinutes(taskLength, taskLengthUnit),
         taskPriority);
-    const taskJson = JSON.stringify(task);
-    allTaskJson.push(taskJson);
+    allTaskJson.push(task);
```
Comment (talvarez0507) [inline portfolio/src/main/webapp/task.js:234]: This is where the bug was. collectAllTasks was returning an array of JSON objects and when information is sent to the servlet it is already converted to JSON so it was unnecessary.

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -74,17 +74,11 @@
...
-                class="nav-link dropdown-toggle"
-                role="button"
-                data-toggle="dropdown"
-                aria-haspopup="true"
-                aria-expanded="false"
-              >
-                Scheduling algorithm preferences
-              </a>
-              <div class="dropdown-menu" aria-labelledby="navbar-dropdown"></div>
+            <li class="nav-item nav-link">
```
Comment (talvarez0507) [inline portfolio/src/main/webapp/index.html]: This is the dropdown where a user can choose their preferred type of scheduling (the algorithm).

**Review by raulcruise — APPROVED** [review]

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (jamesnam-google) [inline portfolio/src/main/webapp/index.html]: Consider putting this dropdown next or above the "Start scheduling!" button

File: portfolio/src/test/java/com/google/sps/CollectEventsFromJsonArrayTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.Collection;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class CollectEventsFromJsonArrayTest {
```
Comment (jamesnam-google) [inline portfolio/src/test/java/com/google/sps/CollectEventsFromJsonArrayTest.java]: I think this file should be named ServletHelperTest. The test file needs to be associated with the Java file.

File: portfolio/src/test/java/com/google/sps/CollectTasksFromJsonArrayTest.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import java.util.Collection;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class CollectTasksFromJsonArrayTest {
```
Comment (jamesnam-google) [inline portfolio/src/test/java/com/google/sps/CollectTasksFromJsonArrayTest.java]: Tests in this file can go into the ServletHelperTest

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -74,17 +74,11 @@
...
+              Scheduling Type:
+              <select id="algorithm-type" name="algorithm-type">
+                <option value="SHORTEST_TASK_FIRST">Shortest tasks first</option>
```
Comment (jamesnam-google) [inline portfolio/src/main/webapp/index.html]: This is okay, but I think the default option should be highest priority tasks first. Consider adding TODO comment.

**Review by jamesnam-google — APPROVED** [review]
Approving the change with comments

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (talvarez0507) [reply portfolio/src/main/webapp/index.html]: I moved it to the left of the button.

File: portfolio/src/test/java/com/google/sps/CollectEventsFromJsonArrayTest.java (hunk 3 above)
Comment (talvarez0507) [reply portfolio/src/test/java/com/google/sps/CollectEventsFromJsonArrayTest.java]: Done.

File: portfolio/src/test/java/com/google/sps/CollectTasksFromJsonArrayTest.java (hunk 4 above)
Comment (talvarez0507) [reply portfolio/src/test/java/com/google/sps/CollectTasksFromJsonArrayTest.java]: Done.

File: portfolio/src/main/webapp/index.html (hunk 5 above)
Comment (talvarez0507) [reply portfolio/src/main/webapp/index.html]: Once other algorithms are implemented they'll be added and we'll change the default option.

**Review by rgcase — APPROVED** [review]

**Pull request merged**